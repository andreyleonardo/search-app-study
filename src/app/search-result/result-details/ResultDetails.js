import React from 'react';
import PropTypes from 'prop-types';
import he from 'he';
import classnames from 'classnames';

import './ResultDetails.scss';

const ResultDetails = ({
  index, description, expand, imgAlt, imgSrc, title
}) => (
  <div
    id={`result-details-id-${index}`}
    aria-expanded={expand}
    className={classnames('result_details row', { 'result_details--expanded': expand })}
  >
    <picture className="result_details__picture hidden-xs col-sm-12">
      <img className="result_details__img" src={imgSrc} alt={imgAlt} />
    </picture>
    <article className="result_details__description col-xs-12">
      <h1>
        {he.decode(title)}
      </h1>
      <p>{he.decode(description)}</p>
    </article>
  </div>
);

ResultDetails.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  expand: PropTypes.bool,
  imgAlt: PropTypes.string,
  imgSrc: PropTypes.string.isRequired,
  index: PropTypes.number
};

ResultDetails.defaultProps = {
  expand: false,
  imgAlt: '',
  index: 0
};

export default ResultDetails;
