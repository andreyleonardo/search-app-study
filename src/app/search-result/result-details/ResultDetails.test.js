import React from 'react';
import ResultDetails from './ResultDetails';

describe('Result Details', () => {
  let resultCard;
  let props;

  beforeEach(() => {
    props = {
      title: 'Result title',
      description: 'Result Description',
      expand: false,
      imgSrc: 'image source'
    };

    resultCard = shallow(<ResultDetails {...props} />);
  });

  it('renders without crashing', () => {
    expect(resultCard).toMatchSnapshot();
  });

  it('sets expanded class when expand is true', () => {
    resultCard.setProps({ expand: true });

    expect(resultCard.exists('.result_details--expanded')).toBeTruthy();
  });
});
