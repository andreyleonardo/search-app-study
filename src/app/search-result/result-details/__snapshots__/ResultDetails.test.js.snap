// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Details renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultDetails
    description="Result Description"
    expand={false}
    imgAlt=""
    imgSrc="image source"
    index={0}
    title="Result title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-expanded": false,
      "children": Array [
        <picture
          className="result_details__picture hidden-xs col-sm-12"
        >
          <img
            alt=""
            className="result_details__img"
            src="image source"
          />
        </picture>,
        <article
          className="result_details__description col-xs-12"
        >
          <h1>
            Result title
          </h1>
          <p>
            Result Description
          </p>
        </article>,
      ],
      "className": "result_details row",
      "id": "result-details-id-0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="result_details__img"
            src="image source"
          />,
          "className": "result_details__picture hidden-xs col-sm-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "result_details__img",
            "src": "image source",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "picture",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Result title
            </h1>,
            <p>
              Result Description
            </p>,
          ],
          "className": "result_details__description col-xs-12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Result title",
            },
            "ref": null,
            "rendered": "Result title",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Result Description",
            },
            "ref": null,
            "rendered": "Result Description",
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-expanded": false,
        "children": Array [
          <picture
            className="result_details__picture hidden-xs col-sm-12"
          >
            <img
              alt=""
              className="result_details__img"
              src="image source"
            />
          </picture>,
          <article
            className="result_details__description col-xs-12"
          >
            <h1>
              Result title
            </h1>
            <p>
              Result Description
            </p>
          </article>,
        ],
        "className": "result_details row",
        "id": "result-details-id-0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="result_details__img"
              src="image source"
            />,
            "className": "result_details__picture hidden-xs col-sm-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "result_details__img",
              "src": "image source",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "picture",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Result title
              </h1>,
              <p>
                Result Description
              </p>,
            ],
            "className": "result_details__description col-xs-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Result title",
              },
              "ref": null,
              "rendered": "Result title",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Result Description",
              },
              "ref": null,
              "rendered": "Result Description",
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
